webapp/src/components/Staking/LiquidityGaugeModal/StakingActionModal.tsx

```diff
@@ -0,0 +1,517 @@
+  const { chainId } = useWeb3Wallet();
+  const { provider } = useWeb3Context();
+  const { data: lg5Data, loading: lg5DataLoading } =
+    useLiquidityGaugeV5PoolData(vaultOption);
+  const { prices, loading: assetPricesLoading } = useAssetsPrice();
+  const {
+    data: { asset, decimals, pricePerShare },
+    loading: vaultDataLoading,
+  } = useV2VaultData(vaultOption);
+
+  const loadingText = useTextAnimation(
+    lg5DataLoading || assetPricesLoading || vaultDataLoading
+  );
+  const { addPendingTransaction } = usePendingTransactions();
+  const [error, setError] = useState<"insufficient_balance">();
+  const [txId, setTxId] = useState("");
+  const vaultContract = useV2Vault(vaultOption);
+  const color = getVaultColor(vaultOption);
+
+  const [baseAPYText] = useMemo(() => {
+    if (lg5DataLoading || assetPricesLoading || vaultDataLoading) {
+      return [loadingText];
+    }
+
+    if (!lg5Data) {
+      return ["0.00%"];
+    }
+
+    const poolRewardInUSD = parseFloat(
+      assetToFiat(lg5Data.poolRewardForDuration, prices["RBN"])
+    );
+    const poolSizeInAsset = lg5Data.poolSize
+      .mul(pricePerShare)
+      .div(parseUnits("1", decimals));
+    const poolSizeInUSD = parseFloat(
+      assetToFiat(poolSizeInAsset, prices[asset], decimals)
+    );
+
+    return [
+      `${
+        poolSizeInUSD > 0
+          ? (((1 + poolRewardInUSD / poolSizeInUSD) ** 52 - 1) * 100).toFixed(2)
+          : "0.00"
+      }%`,
+    ];
+  }, [
+    asset,
+    assetPricesLoading,
+    decimals,
+    lg5Data,
+    lg5DataLoading,
+    loadingText,
+    pricePerShare,
+    prices,
+    vaultDataLoading,
+  ]);
+
+  const handleInputChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const rawInput = e.target.value;
+
+      // Do not allow user to enter smaller than 0
+      if (rawInput && parseFloat(rawInput) < 0) {
+        setInput("");
+        return;
+      }
+
+      setInput(rawInput); // Let's flush out the input changes first.
+    },
+    []
+  );
+
+  const handleMaxPressed = useCallback(() => {
+    if (stakingPoolData) {
+      setInput(formatUnits(stakingPoolData.unstakedBalance, decimals));
+    }
+  }, [decimals, stakingPoolData]);
+
+  const handleClose = useCallback(() => {
+    onClose();
+    if (step === "preview" || step === "walletAction") {
+      setStep("form");
+    }
+    if (step !== "processing") {
+      setInput("");
+    }
+  }, [step, onClose]);
+
+  const handleActionPressed = useCallback(async () => {
+    if (!vaultContract) {
+      return;
+    }
+    setStep("walletAction");
+
+    try {
+      const tx = await vaultContract.stake(parseUnits(input, decimals));
+
+      setStep("processing");
+
+      const txhash = tx.hash;
+
+      setTxId(txhash);
+      addPendingTransaction({
+        txhash,
+        type: "unstake",
+        amount: input,
+        stakeAsset: vaultOption,
+      });
+
+      await provider.waitForTransaction(txhash, 5);
```

Comment:
5 confirmations is too long no? It shows as confirmed with 1 confirmation on etherscan

Reply:
The reason behind 5 is actually consistency across the whole app. 

There is 1 problem that arises when it comes to 1 confirmation, which is if we make a request to get the updated user balance after 1 confirmation after the deposit tx success, the result return is indeterministic where it sometimes shows the balance before the deposit, sometimes after. 

I had tried to only use 5 confirmation when needed and 1 when they are not needed, but it often times result in rather inconsistent UI, (eg toast appear very long after the button no longer shows loading, or toast showed up after more than 10 seconds after the modal closed automatically on success)

So to keep it consistent, I had chosen to wait for 5 confirmations across the whole app so the above problem does not arise.